src/logger.service.ts

```diff
@@ -96,21 +96,14 @@ export class NGXLogger {
       return [];
     }
 
-    const additionalParams: any[] = [];
-
-    for (let i = 0; i < additional.length; i++) {
+    return additional.map((next, idx) => {
       try {
-        if (typeof additional[i] === 'object') {
-          additional[i] = JSON.stringify(additional[i], null, 2);
-        }
-
-        additionalParams.push(additional[i]);
-      } catch (e) {
-        additionalParams.push(`The additional[${i}] value could not be parsed using JSON.stringify().`);
+        return typeof next === 'object' ? JSON.stringify(next, null, 2) : next;
```

Comment:
The reason I suggested using the `typeof next === 'string'` is because you also need to "stringify" any Arrays.

Reply:
It will work both ways, the benefit of `typeof addtional === 'object'` is because it will keep the type of the value when it logs to the console, instead of making everything a string. so console.log('4') is different than console.log(4).